### ISSUE TITLE
Attention mask?

### ISSUE THREAD

**Issue by pcuenca** [huggingface/open-muse#88]
Like in Stable Diffusion, no attention mask appears to be used for input tokens: https://github.com/huggingface/open-muse/blob/2a0365707c0c3d91b0b91bc456161db693652ff1/muse/pipeline_muse.py#L93-L101

 But [according to third party analysis](https://github.com/huggingface/diffusers/pull/2634) this appears to have been a mistake all along. Do we have insight on whether attention masks would help for better prompt-image alignment?

**Comment by Birch-san** [comment]
these authors reckon it's better to train on an unmasked text embeddings (even though that risks learning from PAD token embeddings):  
https://github.com/huggingface/diffusers/issues/1890#issuecomment-1597564866

I thought Muse was a bit wackier though. it actually masks _vision_ tokens:  
<img src="https://raw.githubusercontent.com/lucidrains/muse-maskgit-pytorch/main/muse.png" width="450px">  
https://github.com/lucidrains/muse-maskgit-pytorch/
